api/src/main/java/mod/chiselsandbits/api/placement/PlacementResult.java

```diff
@@ -0,0 +1,80 @@
+package mod.chiselsandbits.api.placement;
+
+import com.mojang.math.Vector4f;
+import mod.chiselsandbits.api.config.IClientConfiguration;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+
+/**
+ * Represents a result of a placement attempt.
+ */
+public class PlacementResult
+{
+
+    /**
+     * Creates a new successful placement result.
+     * @return A new result indicating successful placement.
+     */
+    public static PlacementResult success()
+    {
+        return success(IClientConfiguration.getInstance().getSuccessfulPlacementColor().get());
```

Comment:
This won't work.

PlacementResult is used both the on client as well as on the server side.
As such the client config is not available and even though it does not crash, I consider this bad code and should be avoided.

You might want to modify the API then so that the color can be retrieved from the placement executor (item, blockitem etc) instead of from here, so that when the client code to render the outline is called it can do that on the render thread with the client config loaded.

The other option is to use a DistExecutor call and have the server side have some soft of default value.

Reply:
Ok, so so the `IClientConfiguration` class can be referenced on the server because its in the API, but `IClientConfiguration#getInstance` (or anything else in `IClientConfiguration`, for that matter) should never be called on the server, right?

Reply:
Alright, I used `DistExecutor`.

Reply:
> Ok, so so the `IClientConfiguration` class can be referenced on the server because its in the API, but `IClientConfiguration#getInstance` (or anything else in `IClientConfiguration`, for that matter) should never be called on the server, right?

Yep that is accurate